lektor/packages.py

```diff
@@ -103,6 +103,22 @@ def download_and_install_package(
 
 def install_local_package(package_root, path):
     """This installs a local dependency of a package."""
+
+    # Because of these bugs:
+    # - pip https://github.com/pypa/pip/issues/4390
+    # - setuptools https://github.com/pypa/setuptools/issues/392
+    # we cannot just call `pip install --target $folder --editable $package`.
+    # Hence the workaround of first installing only the package and then it's dependencies
+
+    editable_install_without_dependencies(package_root, path)
+
+    for requirements_path in requirements_from_unfinished_editable_install_at_path(
```

Comment:
I think using a for-loop here is a slight abuse of notation.

Reply:
That is true, but I do need a way to clean up the temporary files afterwards. I could use a context manager for that, but then the empty case (i.e. no requirements) gets more complicated and needs another in band signal to not call installation there.

Not quite sure what the best way forward here would be. I could pull out the whole tempdir into the surrounding function and handle it there, that would clean it up - but also pollute the outer function with a concern it doesn't really need to know about.

Reply:
For loop is gone in the current iteration.